web/server/codechecker_server/api/report_server.py

```diff
@@ -2861,6 +2863,187 @@ def getCheckerCounts(self, run_ids, report_filter, cmp_data, limit,
+                                ),
+                                True
+                            )
+                        ],
+                        else_=False
+                    ).label("isOpened"),
+                    func.count(subquery.c.bug_id)
+                )
+                .group_by(
+                    subquery.c.checker_id,
+                    subquery.c.checker_name,
+                    subquery.c.analyzer_name,
+                    subquery.c.severity,
+                    subquery.c.run_id,
+                    case(
+                        [
+                            (
+                                subquery.c.detection_status.in_(list(map(
+                                    detection_status_str,
+                                    (DetectionStatus.OFF,
+                                     DetectionStatus.UNAVAILABLE)
+                                ))),
+                                False
+                            )
+                        ],
+                        else_=True
+                    ),
+                    case(
+                        [
+                            (
+                                and_(
+                                    subquery.c.detection_status.in_(list(map(
+                                        detection_status_str,
+                                        (DetectionStatus.NEW,
+                                         DetectionStatus.UNRESOLVED,
+                                         DetectionStatus.REOPENED)
+                                    ))),
+                                    subquery.c.review_status.in_(list(map(
+                                        review_status_str,
+                                        (API_ReviewStatus.UNREVIEWED,
+                                         API_ReviewStatus.CONFIRMED))))
+                                ),
+                                True
+                            )
+                        ],
+                        else_=False
+                    )
+                )
+            )
+
+            checker_stats = {}
+            all_run_id = [runId for runId, _ in max_run_histories.all()]
+            for checker_id, \
+                checker_name, \
+                analyzer_name, \
+                severity, \
+                run_ids, \
+                is_enabled, \
+                is_opened, \
+                cnt \
+                    in query.all():
+                checker_stat = checker_stats[checker_id] \
+                    if checker_id in checker_stats \
+                    else CheckerInfo(
+                        checkerName=checker_name,
+                        analyzerName=analyzer_name,
+                        enabled=[],
+                        disabled=all_run_id,
+                        severity=severity,
+                        closed=0,
+                        outstanding=0
+                    )
```

Comment:
What is the expected value for the saturation of the dict's keys, and whether this code takes the true or the false branch? It is said that in Python it's not nice to do `X = D[K] if K in D else V` because it will do **two** lookups in the key set. Depending on whether we expect the key to be already there or not, it's better to use either

**exceptions**

```py
try:
  X = d[K]
except KeyError:
  X = V
```

or the null-coalescing **get()**: `X = D.get(K, V)`